### ISSUE TITLE
pgp tagging

### ISSUE THREAD

**Issue by timo-schaefer** [allegro/axion-release-plugin#366]
Hi,

I'm using axion-release-plugin 1.12.1 which uses jgit 5.7.0 internally.
Now I'm trying to sign the version tag with gpg created by the plugin.

My .gitconfig looks like this:
``` 
[user]
	signingkey = 531BFF72
[commit]
	gpgsign = true
[tag]
	gpgsign = true
``` 
The tag which has been created by the plugin has not been signed and it looks like that the git configuration has no effect and has been ignored.

**Comment by bgalek** [comment]
Hi @timo-schaefer!
I'm going to update jgit (I also had problems with signing commits, but mine was resolved just by updating) https://github.com/allegro/axion-release-plugin/pull/370 
I'll get back to you with a snapshot release to test, ok? :)

**Comment by timo-schaefer** [comment]
Hi,
perfect. I'm looking forward to test your snapshot...

**Comment by bgalek** [comment]
@timo-schaefer could you try-it yourself simply by building master and using your local m2?
It's more secure then me sending You a jar file ;)

**Comment by timo-schaefer** [comment]
I have build master and I can see the plugin in my local maven repository, but I'm not able to include into my project.
I have also read your _DEVELOPMENT.md_ file and changed my _build.gradle_ accordingly but the plugin is not found.
I get the following error:
```
Plugin with id 'pl.allegro.tech.build.axion-release' not found
```
I'm using gradle wrapper version 6.8.1
When I do a different approach, modifying my _settings.gradle_ like this:
```
pluginManagement {
    repositories {
        mavenLocal()
	gradlePluginPortal()
    }
}
```
and using the plugin id in the build.gradle `id 'pl.allegro.tech.build.axion-release' version '1.12.2-SNAPSHOT'` , I get the following error:
```
Build file '/home/tschaefer/sources/my-core-service/build.gradle' line: 16

* What went wrong:
Plugin [id: 'pl.allegro.tech.build.axion-release', version: '1.12.2-SNAPSHOT'] was not found in any of the following sources:

- Gradle Core Plugins (plugin is not in 'org.gradle' namespace)
- Plugin Repositories (could not resolve plugin artifact 'pl.allegro.tech.build.axion-release:pl.allegro.tech.build.axion-release.gradle.plugin:1.12.2-SNAPSHOT')
  Searched in the following repositories:
    MavenLocal(file:/home/tschaefer/.m2/repository/)
    Gradle Central Plugin Repository
```
I also tried it with the following configuration:
```
pluginManagement {
    repositories {
        mavenLocal()
		gradlePluginPortal()
    }
    resolutionStrategy {
        eachPlugin {
            if (requested.id.namespace == 'pl.allegro.tech.build') {
                useModule("pl.allegro.tech.build:axion-release-plugin:${requested.version}")
            }
        }
    }
}
```
but then I get the following error:
```
Could not apply requested plugin [id: 'pl.allegro.tech.build.axion-release', version: '1.12.2-SNAPSHOT', artifact: 'pl.allegro.tech.build:axion-release-plugin:1.12.2-SNAPSHOT'] as it does not provide a plugin with id 'pl.allegro.tech.build.axion-release'. This is caused by an incorrect plugin implementation. Please contact the plugin author(s).
> Plugin with id 'pl.allegro.tech.build.axion-release' not found.
```
I would love to test, but after hours of getting this to work and searching in the web, I'm giving up...

**Comment by bgalek** [comment]
@timo-schaefer thaks for your effort! I'll update DEVELOPMENT.md, but for now - is your project opensourced? maybe I could simply test your project with my local plugin snapshot?

**Comment by timo-schaefer** [comment]
With gradle wrapper 5.6.4 I was able to apply the locally build plugin. But unfortunately the tag created by the task _release_ still has no signature. If I tag a commit manually it has a signature.

**Comment by bgalek** [comment]
@timo-schaefer thank you! I'll check this out right away!

**Comment by bgalek** [comment]
@timo-schaefer I've looked around jgit code and I think that we need this commit to be merged https://git.eclipse.org/c/jgit/jgit.git/commit/?id=41b9159795d6f64bba6a67ce2f22fe1b7679ea55
It's going to be released in 5.11.0. We could try with the nightly build (5.11.0-SNAPSHOT) to see if the problem will be resolved. Before the 5.11 release, we would have to find a workaround. I'll try to create a test case to document current behavior.

**Comment by timo-schaefer** [comment]
Do we have any progress regarding this issue?
It would really be great if it is possible to sign the tags.
